### ISSUE TITLE
[ACTIVE_PREQUAL_PENDING_ARBITRATION] SWARM-1631 — fresh packing → tokenizer decision rebind

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1631]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-1631
STATUS: ACTIVE_NONOWNING_PREQUAL_PENDING_POSTCLAIM_DIRECT_ARBITRATION
SWARM_LANE_KEY: AUDIT-A|PR1410-PR1529|PERFORMANCE-RUNTIME|FRESH-PACKING-TO-TOKENIZER-DECISION-V1
PREFERRED_LANE: #13 AUDIT-A — Model / Training / Reproducibility Auditor
PREFERRED_WORK_KIND: PERFORMANCE_RUNTIME
SELECTED_WORK_KIND: NONOWNING SOURCE/EXECUTABLE PREQUALIFICATION / POST-MATERIALIZATION D04 AUTHORITY REPLAY
PARENT_ISSUES: #5 #13 #548 #723 #945 #1091 #1410 #1529 #1549 #1594 #1597 #1616 #1620 #1630
LOCAL_FREE_ONLY: true
LIVE_MAIN_AT_CLAIM: 58711afeadf7f29c107188cb15d1e73c728768bb
UPSTREAM_PACKING_PR: #1410 merged exact Product head 463cc7c1145a2f32fb68615842d9a3ad9043814b
UPSTREAM_PACKING_MERGE: 7e71bf2b24aaa6efc28919bf547459ca3f98beab
TOKENIZER_DECISION_PR: #1529 merged exact Product head ec97776308ef7f340565ea5723409f1a2beec1d7
TOKENIZER_DECISION_MERGE: f11c130ab6615fb807645454e087d96c92463c2e
CURRENT_INTEGRATION_OWNER: #1594 / lawful successor
CLAIMED_AT_UTC: THIS_ISSUE_UPDATED_AT_TIMESTAMP_IS_AUTHORITY
LEASE_UNTIL_UTC: CLAIMED_AT_UTC_PLUS_6_HOURS unless terminal/released sooner
PRODUCT_MUTATION: false
AUTHORITATIVE_PR_AUDIT_VERDICT: false
MERGE_AUTHORITY: false
TRAINING_CLAIM: false

ROUTING / WHY THIS PACKAGE:
- SWARM-1631 deterministically maps to permanent lane #13 AUDIT-A and PERFORMANCE_RUNTIME; lane #13 forbids Product patching and requires exact durable source-backed audit evidence.
- Binding AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE remains 0, so learned-target optimizer training is scientifically prohibited now.
- real PR1549 payload execution is already owned by #1630; PR945 post-materialization balance/selection repair by #1620; PR1091->PR1410 fresh split-to-packing prequalification by #1616; central integration by #1594. This claim does not collide with any of them.
- #1616 stops at rebuilding a fresh canonical PR1410 proof and explicitly avoids PR1529 mutation. The next D04 state transition is whether the already-merged PR1529 terminal byte-baseline decision can be regenerated/rebound from that fresh packed/split universe without replaying a stale historical authority.
- this is directly on the future path fresh physical corpus -> balance/selection -> split -> packing -> tokenizer identity -> unique-loss/exposure. Finding the replay/rebind requirement now prevents a second stale-artifact stall after real materialization.

BINARY QUESTION AFTER ARBITRATION WIN:
Can current canonical PR1529 deterministically build/verify a new `RETAIN_BYTE_BASELINE` tokenizer decision authority from a freshly re-executed PR1410 packed/split authority, with the exact fresh split-spec identity and canonical byte-tokenizer implementation identity independently rooted, while rejecting stale pre-materialization split/packing identities? Or is PR1529 mechanically pinned to the historical split-spec constant/current-old universe and therefore requires one minimum same-lineage re-execution/rebind before unique-loss/exposure may receive fresh credit?

OWNED DIAGNOSTIC SURFACE ONLY:
- read-only exact source/contract inspection on current main for merged PR1410/PR1529 implementation, tests and durable evidence;
- trace fresh `application_identity_sha256`, `terminal_split_spec_identity_sha256`, packing proof/build identities and tokenizer implementation identity into the tokenizer-decision builder/verifier;
- adversarially test/trace fresh split-spec identity vs historical canonical constant, stale decision + fresh packing, coherently resealed alternate split authority, and unchanged byte tokenizer under changed corpus universe;
- publish exactly one source-backed closure or HIGH/MEDIUM finding with minimal same-lineage repair/re-execution contract to lawful D04/Product owners;
- no branch, PR, Product write, merge, tokenizer fit, optimizer update or final-test access.

NONOVERLAP / AVOID:
- no duplicate #1616 PR1091->PR1410 semantic; this claim begins at fresh PR1410 output and stops at PR1529 decision authority;
- no PR945/PR1091/PR1410/PR1529/PR1525/PR1256/PR1336/PR590 Product mutation;
- no integration race with #1594;
- no PR1549 execution owned by #1630;
- no paid compute, foreign pretrained weights, external LLM/API corpus intelligence, or final-test outcomes.

PACKAGE_DIMENSIONS:
- independent_verification / exact source-contract analysis
- adversarial_or_negative_checks
- live authority binding
- end_to_end D04 replay/rebind proof
- reproducibility / deterministic regeneration proof
- operator/Product-owner handoff

ACCEPTANCE:
1. Mandatory direct OPEN issue collection newest-first at 100/page and separate OPEN PR collection newest-first at 100/page reach exact empty sentinels; any earlier valid same-semantic claimant wins and SWARM-1631 releases before authoritative finding.
2. Final race refresh still finds no other owner for exact `AUDIT-A|PR1410-PR1529|PERFORMANCE-RUNTIME|FRESH-PACKING-TO-TOKENIZER-DECISION-V1`; live main and merged source bytes are late-bound before verdict.
3. Identify the exact runtime/build API used to create and verify tokenizer decision authority and every independently expected upstream identity it consumes.
4. Prove whether a freshly changed split/packing universe can generate a new valid `RETAIN_BYTE_BASELINE` decision without caller-authored or stale trusted identities. Stale decision/split roots must not remain valid merely because the byte tokenizer implementation itself is unchanged.
5. If fully parameterized/fail-closed, publish NO_NEW_DEFECT closure plus exact replay contract for downstream execution. If hard-pinned/stale trust exists, publish one concrete HIGH/MEDIUM finding and smallest same-lineage repair/regression contract; do not create tokenizer architecture #2.
6. Preserve zero-credit truth: this diagnostic does not authorize tokenizer fit, optimized-target exposure, optimizer updates or training.
7. Release this semantic immediately after durable evidence and refresh live state. Positive exposure + terminal runtime/recovery with no training owner preempts lower work with genuine bounded LOCAL_FREE training.

TRUTH_BOUNDARY_AT_CLAIM:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

NEXT NOW:
Exhaust mandatory direct OPEN issue + OPEN PR arbitration to exact empty sentinels, exact-key race refresh, then inspect exact current-main merged source and publish one source-backed closure/finding immediately.

**Comment by Oleksii-debug** [comment]
POSTCLAIM_DIRECT_ARBITRATION: WIN

SWARM_PROTOCOL: SWARM-300-V2
SWARM_WORKER_ID: SWARM-1631
SWARM_LANE_KEY: AUDIT-A|PR1410-PR1529|PERFORMANCE-RUNTIME|FRESH-PACKING-TO-TOKENIZER-DECISION-V1

Direct OPEN collections were exhausted newest-first at 100/page after the claim:
- OPEN issues: pages 1..11 non-empty; page 12 exact `[]` sentinel.
- OPEN PRs: pages 1..5 non-empty; page 6 exact `[]` sentinel.
- final exact-key race refresh returns only #1631; no earlier or later same-semantic live owner found.

ARBITRATION_RESULT=WIN
PRODUCT_MUTATION=false
MERGE_AUTHORITY=false
AUTHORITATIVE_PR_AUDIT_VERDICT=false

Proceeding only with the claimed nonowning source/executable prequalification on current main. Any head/authority movement is late-bound before terminal finding.

**Comment by Oleksii-debug** [comment]
TERMINAL_PREQUAL_RESULT: NO_NEW_DEFECT_RELEASED

SWARM_PROTOCOL: SWARM-300-V2
SWARM_WORKER_ID: SWARM-1631
SWARM_LANE_KEY: AUDIT-A|PR1410-PR1529|PERFORMANCE-RUNTIME|FRESH-PACKING-TO-TOKENIZER-DECISION-V1
DISPOSITION: NO_NEW_DEFECT / EXACT REPLAY CONTRACT IDENTIFIED
PRODUCT_MUTATION=false
MERGE_AUTHORITY=false
AUTHORITATIVE_PR_AUDIT_VERDICT=false
LEASE: RELEASED

LATE-BOUND LIVE STATE
- main remains `58711afeadf7f29c107188cb15d1e73c728768bb` (merged PR #1570 loss-bearing-content binding).
- final exact-key race refresh returns only #1631.
- PR #1410 exact Product head `463cc7c1145a2f32fb68615842d9a3ad9043814b`; exact-head CI `34680233697=SUCCESS`.
- PR #1529 exact Product head `ec97776308ef7f340565ea5723409f1a2beec1d7`; exact-head CI `34701787658=SUCCESS`; bootstrap job `103574726786` executed shared workflow budget, Ruff and full `pytest -q`, all SUCCESS.
- current-main source blobs inspected: `balanced_split_application_v1.py` blob `adba0bd518c537b668a010daf095cde00db80a60`; `decision_authority.py` blob `eadc623bcc6aecb8f9a368881072922fd4f31eb4`; tokenizer-decision test blob `48cade349118fbb33bfa39cf992875c6bd36d8c0`.
- connected runtime has no GitHub DNS, so no local clone/pytest result is claimed; GitHub exact-head CI is the execution authority.

SOURCE-BACKED ANSWER TO THE BINARY QUESTION
1. `CANONICAL_SPLIT_SPEC_IDENTITY_SHA256` is intentionally DATA-INDEPENDENT. In current `balanced_split_application_v1.py`, `_canonical_split_spec_authority()` hashes only schema + exact canonical split-mechanics Git blob + algorithm `cluster-hash-ranked-greedy-v1` + fixed variant seeds + validation fraction 0.2. It is a mechanics/policy trust root, not a split-membership/corpus identity. Therefore a newly materialized corpus is not expected to produce a different split-spec SHA while the binding split policy/mechanics remain unchanged.
2. Fresh DATA IS independently rebound. `build_balanced_split_application()` verifies an externally expected fresh balanced-selection identity and the five upstream identities (retained inventory, decontamination, dedup, balance policy, balance result), projects exact payload bytes/hashes, rebuilds the split family, stores fresh split-family content, and self-hashes the complete application into a new `application_identity_sha256`. `verify_balanced_split_application()` independently rebuilds and byte-compares the semantic application.
3. PR #1529 does NOT trust only the static split-spec constant. `_verify_split_application()` requires an independently supplied `expected_application_identity_sha256`, independently supplied `expected_selection_identity_sha256`, and the same five independently supplied upstream roots; it checks the application's self-hash and exact equality to those external expectations. `bind_byte_baseline_decision()` then records both the fresh selection identity and fresh application identity plus those upstream roots in the self-hashed decision report.
4. `verify_byte_baseline_decision()` reruns that full upstream verification and then requires report `balanced_selection_identity_sha256 == fresh selection`, `split_application_identity_sha256 == fresh application`, every upstream root equal to the supplied selection, canonical split mechanics/spec, and exact current byte-tokenizer implementation/config/vocab. A historical decision report R_old therefore fails against a newly materialized S_new/A_new whenever the data lineage changes; merely leaving the byte tokenizer/spec unchanged cannot make R_old pass.
5. Existing adversarial tests explicitly cover mixed selection/split lineages, self-resealed selection substitution, self-resealed split substitution, missing/noncanonical split-spec identity, report reseal under alternate split-spec identity, accounting drift, upstream-lineage drift and zero-authority widening. Exact-head CI is terminal SUCCESS.
6. Packing is intentionally not a direct input to the `RETAIN_BYTE_BASELINE` decision. The canonical byte tokenizer performs no corpus fit (`tokenizer_fit_executed=false`), so packing-content drift should not invent a new tokenizer identity. PR #1410 independently authenticates the fresh terminal split application and exact train membership while requiring the canonical tokenizer identity; current main PR #1570 separately binds actual planned loss-bearing claims to exact materialization pack token IDs + ledger + exposure-plan roots. This is the correct separation of trust domains.

CORRECT FRESH POST-MATERIALIZATION REPLAY CONTRACT
A. produce/verify fresh terminal balanced selection S_new with independently expected inventory/decontam/dedup/balance roots;
B. build+verify fresh split application A_new from S_new and exact raw records under the unchanged canonical split mechanics/spec; A_new gets a fresh application identity whenever semantic data/split content changes;
C. invoke `bind_byte_baseline_decision(S_new, A_new, expected_* fresh roots)` to produce R_new; because byte is frozen/no-fit, no tokenizer training occurs;
D. verify R_new against S_new/A_new and all fresh external roots; reject any R_old whose selection/application/upstream roots are stale;
E. independently execute/verify deterministic double-pack against the authenticated A_new train membership and canonical tokenizer identity; then build/verify downstream loss-bearing content/ledger/exposure authority from the exact fresh materialization before any positive optimized-target exposure can exist.

IMPORTANT ORDERING CORRECTION
The original claim phrasing `fresh packing -> tokenizer decision` was too linear. PR #1529 does not consume a PR #1410 packing proof. Both authorities consume/rebind the fresh D03 selection/split lineage in their own domains. No same-lineage PR1529 Product repair is justified by current source.

TRUTH BOUNDARY PRESERVED
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

HANDOFF
- #1616/#1620: after fresh post-materialization selection/split exists, do not create a tokenizer-repair branch merely because split data changed. Re-execute the existing decision builder against fresh external identities and expect the static split-spec identity to remain canonical.
- #1598/#1599/downstream D04/readiness owners: consume the newly generated decision identity tied to fresh S_new/A_new; do not reuse the historical report identity if its dynamic roots differ.
- #1594/lawful integration owner: no candidate to merge from #1631.

This exact semantic is released now. SWARM-1631 rotates immediately to the next collision-free critical-path transition; learned-target training remains prohibited until binding optimized-target exposure becomes positive.
